syntax = "proto3";

package ottx.otttoken;

option java_package = "code.comcast.ottx.otttoken";
option java_multiple_files = true;

option go_package = "github.com/comcast-ottx-racers/ottx-protobuf/api/ott_token;ott_token";

// OttToken handles Silent Token retrival per Partner
service OttTokenService {
    // AuthToken request to retrieve the token per partner.
    rpc AuthToken (AuthTokenRequest) returns (AuthTokenResponse);
    rpc PlatformToken (PlatformTokenRequest) returns (PlatformTokenResponse) ;
}

// AuthToken request to retrieve the token per partner.
// SAT Token could be passed in the Authorization header
message AuthTokenRequest {
    // Partner Id.
    string partner_id = 1;
    // Application Id .
    string app_id = 2;
    // XSCT token.
    string xsct = 3;
    // SAT token.
    string sat = 4;
}

message AuthTokenResponse {
    // Token identifying the device to a user
    string access_token = 1;
    // Token type provided by the activation server
    string token_type = 2;
    // provides the scope of the token
    string scope = 3;
    // Numerical value in seconds denoting expiry
    int32 expires_in = 4;
}
// PlatformTokenRequest request to request a platform (CIMA) token.
// SAT Token must also be passed in the Authorization header (and validated against sat in payload)
message PlatformTokenRequest {
    // XACT token.
    string xact = 1;
    // SAT token.
    string sat = 2;
    // DurableAppId for mapping with cima_delegated_client_id.
    string app_id = 3;
}
// Response containing platform (CIMA) token
// TODO - do we also need a refresh message/refresh callback url ?
message PlatformTokenResponse {
    //the token
    string platform_token = 1;
    //token type - will only be "cima" for now, but adding this here for future proofingoi
    string token_type = 2;
    // Numerical value in seconds denoting expiry
    int32 expires_in = 3;
    // provides the scope of the token
    string scope = 4;
}